packages/host/app/resources/file.ts

```diff
@@ -103,11 +100,12 @@ class _FileResource extends Resource<Args> {
     if (this.onStateChange && this.innerState.state !== prevState.state) {
       this.onStateChange(this.innerState.state);
     }
-
-    if (newState.state === 'ready') {
-      this.recentFilesService.addRecentFile(newState.url);
```

Comment:
@backspace I removed this. Thought that maybe recentFiles wud just interact with codePath.

Reply:
Hmmm ahh I see the difference of having this. It eagerly adds the file. Otherwise, the first file is not added to recent files. Ok I can easily add this back in. But I will add it to top-level code mode

Reply:
yeah, i added this so that we have a central (and deep) place that we keep track of files that have been opened (before this, this logic was duplicated in multiple places because it wasn't deep enough). I'd like to keep this logic located in the file resource, but we can add a file to recent items after we get the response from the server (which may include a redirect)